gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -44,31 +43,32 @@
  * Contains the methods that executes in spark executor for {@link
  * GCImpl#identifyExpiredContents(SparkSession)}.
  */
-public class IdentifyContentsPerExecutor implements Serializable {
+public class IdentifyContentsPerExecutor {
 
-  private final GCParams gcParams;
```

Comment:
I've removed the state from this class and changed processing to `static` methods.
This is a favorable way to do processing in spark as it reduces the number of code that needs to be serialized.

Reply:
not sure I fully understand why moving things to `static` reduces the amount of code that needs to be serialized. Can you elaborate here a bit please?

Reply:
When we have a spark processing and there is an instance variable on the spark class (state on the class) the whole class needs to be serialized and sent by the spark to executors (https://stackoverflow.com/questions/40818001/understanding-spark-serialization) - for that reason, it is a common pattern in spark to have stateless processing to avoid the need of serialization.